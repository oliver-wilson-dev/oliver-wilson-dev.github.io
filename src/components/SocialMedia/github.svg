<?xml version="1.0" encoding="UTF-8"?>
<svg viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <defs></defs>
    <g id="icons" stroke="none" stroke-width="1">
        <g id="Artboard" transform="translate(-727.000000, -100.000000)">
            <path d="M738.998895,100 C732.373277,100 727,105.372541 727,112.000368 C727,117.302189 730.438043,121.799564 735.206513,123.386353 C735.8069,123.496854 736.025691,123.126308 736.025691,122.808067 C736.025691,122.523712 736.015378,121.768624 736.009485,120.767488 C732.671629,121.492372 731.967372,119.158599 731.967372,119.158599 C731.421499,117.772185 730.634734,117.403112 730.634734,117.403112 C729.545198,116.659075 730.717241,116.673808 730.717241,116.673808 C731.921698,116.758525 732.555235,117.910679 732.555235,117.910679 C733.625618,119.744252 735.364161,119.214586 736.047792,118.907394 C736.156819,118.132417 736.466957,117.603487 736.809509,117.303662 C734.144971,117.00089 731.343411,115.971024 731.343411,111.372725 C731.343411,110.062924 731.811197,108.991068 732.578808,108.152736 C732.455048,107.849228 732.043249,106.628564 732.696676,104.976948 C732.696676,104.976948 733.703705,104.654287 735.996225,106.207189 C736.95316,105.940514 737.980079,105.807913 739.000368,105.802756 C740.019921,105.807913 741.046103,105.940514 742.004512,106.207189 C744.295558,104.654287 745.301114,104.976948 745.301114,104.976948 C745.956015,106.628564 745.544216,107.849228 745.421192,108.152736 C746.190276,108.991068 746.654379,110.062924 746.654379,111.372725 C746.654379,115.982811 743.848399,116.997207 741.175757,117.294085 C741.605973,117.664631 741.989779,118.396881 741.989779,119.516621 C741.989779,121.120354 741.975045,122.414684 741.975045,122.808067 C741.975045,123.129255 742.191627,123.502747 742.800117,123.385617 C747.564904,121.795144 751,117.300715 751,112.000368 C751,105.372541 745.626723,100 738.998895,100" id="github"></path>
        </g>
    </g>
</svg>