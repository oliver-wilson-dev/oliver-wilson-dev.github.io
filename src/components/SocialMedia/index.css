.socialLogo {
    height: 35px;
    width: 35px;
    fill: var(--primaryTextColorInverse);
    transition-property: transform;
    transition-timing-function: ease;
    transition-duration: 0.5s;
    
    &:hover {
        transform: scale(1.2);
    }


}

.socialContainer {
    display: flex;
    flex-direction: row;
    margin: 10px;
}

.socialLink {
    margin-right: 30px;
    &:last-child {
        margin-right: 0px;
    }
}

@media (min-width: 0px) and (max-width: 767px) {
    .socialContainer {
        flex-direction: column;
        flex-basis: 50%;
        align-items: flex-end;
    }

    .socialLink {
        margin-right: 0px;
        margin-bottom: 10px;
        &:last-child {
            margin-bottom: 0px;
        }
    }
  }