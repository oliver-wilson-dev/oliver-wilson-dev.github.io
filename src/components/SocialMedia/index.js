import React from 'react';
import cn from 'classnames';
import styles from './index.css';
import sharedStyles from '../App/index.css';
import LinkedIn from './linkedin.svg';
import Github from './github.svg';
import Phone from './phone.svg';
import Email from './email.svg';

const SocialMedia = () => (
  <div className={cn(styles.socialContainer, sharedStyles.flexCenter)}>
    <a className={styles.socialLink} href="https://www.linkedin.com/in/oliver-wilson-profile/" rel="noopener noreferrer" target="_blank">
      <LinkedIn className={styles.socialLogo} />
    </a>
    <a className={styles.socialLink} href="https://github.com/oliver-wilson-dev/" rel="noopener noreferrer" target="_blank">
      <Github className={styles.socialLogo} />
    </a>
    <a className={styles.socialLink} href="tel:07580081310">
      <Phone className={styles.socialLogo} />
    </a>
    <a className={styles.socialLink} href="mailto:contact@oliverwilson.dev">
      <Email className={styles.socialLogo} />
    </a>
  </div>
);

export default SocialMedia;
