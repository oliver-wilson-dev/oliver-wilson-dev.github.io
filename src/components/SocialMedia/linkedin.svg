<svg viewBox="0 0 18 18" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <defs></defs>
    <g id="icons" stroke="none" stroke-width="1">
        <g id="Artboard" transform="translate(-424.000000, -103.000000)">
            <path d="M426.144077,103 C427.326538,103 428.286769,103.960231 428.286769,105.143385 C428.286769,106.327231 427.326538,107.287462 426.144077,107.287462 C424.956769,107.287462 424,106.327231 424,105.143385 C424,103.960231 424.956769,103 426.144077,103 L426.144077,103 Z M424.293538,108.912308 L427.993231,108.912308 L427.993231,120.806154 L424.293538,120.806154 L424.293538,108.912308 Z M430.230769,108.833385 L433.774,108.833385 L433.774,110.459615 L433.824538,110.459615 C434.317462,109.524308 435.524154,108.538462 437.322769,108.538462 C441.064692,108.538462 441.756308,111.001 441.756308,114.204308 L441.756308,120.727231 L438.060769,120.727231 L438.060769,114.943 C438.060769,113.563923 438.036538,111.789538 436.139615,111.789538 C434.216385,111.789538 433.922846,113.292538 433.922846,114.844692 L433.922846,120.727231 L430.230769,120.727231 L430.230769,108.833385 Z" id="linkedin"></path>
        </g>
    </g>
</svg>