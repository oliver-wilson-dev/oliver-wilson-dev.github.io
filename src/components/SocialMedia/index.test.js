import React from 'react';
import { shallow } from 'enzyme';
import SocialMedia from '.';

const render = () => shallow(<SocialMedia />);

describe('<SocialMedia/>', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });
});
