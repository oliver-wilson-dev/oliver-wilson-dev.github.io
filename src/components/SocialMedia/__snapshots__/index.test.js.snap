// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialMedia/> should render correctly 1`] = `
<div
  className="socialContainer flexCenter"
>
  <a
    className="socialLink"
    href="https://www.linkedin.com/in/oliver-wilson-profile/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <LinkedIn
      className="socialLogo"
    />
  </a>
  <a
    className="socialLink"
    href="https://github.com/oliver-wilson-dev/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <Github
      className="socialLogo"
    />
  </a>
  <a
    className="socialLink"
    href="tel:07580081310"
  >
    <Phone
      className="socialLogo"
    />
  </a>
  <a
    className="socialLink"
    href="mailto:contact@oliverwilson.dev"
  >
    <Email
      className="socialLogo"
    />
  </a>
</div>
`;
